### PR TITLE
Update services_for_ios.rst

### PR DESCRIPTION
**Pull request by Xioor**
Proposing these changes since there is missing documentation on finalizing a purchase for iOS. https://github.com/godotengine/godot-docs/issues/2715

### CODE REVIEW

File: tutorials/platform/services_for_ios.rst [hunk 1]
```diff
@@ -70,6 +70,8 @@ has three methods for purchasing:
 -  ``Error purchase(Variant p_params);``
 -  ``Error request_product_info(Variant p_params);``
 -  ``Error restore_purchases();``
+-  ``void  set_auto_finish_transaction(bool b);``
+-  ``void  finish_transaction(String product_id);``
```
Comment (Calinou) [inline tutorials/platform/services_for_ios.rst:74]:
```suggestion
-  ``void set_auto_finish_transaction(bool b);``
-  ``void finish_transaction(String product_id);``
```

File: tutorials/platform/services_for_ios.rst [hunk 2]
```diff
@@ -81,7 +83,9 @@ and the pending_event interface
 purchase
 ~~~~~~~~
 
-Purchases a product id through the Store Kit API.
+Purchases a product id through the Store Kit API. You have to call finish_transaction(product_id) once you
```
Comment (Calinou) [inline tutorials/platform/services_for_ios.rst:86]:
```suggestion
Purchases a product ID through the Store Kit API. You have to call finish_transaction(product_id) once you 
```

File: tutorials/platform/services_for_ios.rst [hunk 3]
```diff
@@ -81,7 +83,9 @@ and the pending_event interface
 purchase
 ~~~~~~~~
 
-Purchases a product id through the Store Kit API.
+Purchases a product id through the Store Kit API. You have to call finish_transaction(product_id) once you 
+receive a successful response or call set_auto_finish_transaction(true) prior to calling purchase.
```
Comment (Calinou) [inline tutorials/platform/services_for_ios.rst:87]:
```suggestion
receive a successful response or call ``set_auto_finish_transaction(true)`` prior to calling purchase. 
```

File: tutorials/platform/services_for_ios.rst [hunk 4]
```diff
@@ -169,6 +173,38 @@ The response events will be dictionaries with the following fields:
       "result": "ok",
       "product id": "product id of restored purchase"
     }
+    
+set_auto_finish_transaction
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If set to true, once a purchase is successful, your purchase will be 
+finalized automatically. Call this method prior to calling purchase.
```
Comment (Calinou) [inline tutorials/platform/services_for_ios.rst:181]:
```suggestion
If set to ``true``, once a purchase is successful, your purchase will be 
finalized automatically. Call this method prior to calling ``purchase()``.
```

File: tutorials/platform/services_for_ios.rst [hunk 5]
```diff
@@ -169,6 +173,38 @@ The response events will be dictionaries with the following fields:
...
+~~~~~~~~~~~~~~~~~~
+
+If you don't want transactions to be automatically finalized, call this
+method after you receive a successful purchase response.
+
+
+Parameters
+^^^^^^^^^^
+Takes a string as a parameter, with one field, product_id, a string 
+specifying what product to finalize the purchase on. Example:
```
Comment (Calinou) [inline tutorials/platform/services_for_ios.rst:203]:
```suggestion
Parameters
^^^^^^^^^^

Takes a string ``product_id`` as an argument. ``product_id`` specifies what product to finalize the purchase on. Example:
```

File: tutorials/platform/services_for_ios.rst [hunk 6]
```diff
@@ -169,6 +173,38 @@ The response events will be dictionaries with the following fields:
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If set to true, once a purchase is successful, your purchase will be 
+finalized automatically. Call this method prior to calling purchase.
+
+Parameters
+^^^^^^^^^^
+
+Takes a bool as a parameter, with one field, b, a bool specifying if 
+purchases should be automatically finalized. Example:
```
Comment (Calinou) [inline tutorials/platform/services_for_ios.rst:187]:
```suggestion
Takes a boolean ``b`` as a parameter which specifies if purchases should be
automatically finalized. Example:
```

**Review by akien-mga — APPROVED** [review]

**Comment by Difference** [comment]
Please merge this. There are important information on how to finish IOS transactions in it.

**Comment by pycbouh** [comment]
@Difference It has been merged months ago, but only for the latest documentation. Does it need to be picked for 3.4, 3.3?

**Comment by Difference** [comment]
> @Difference It has been merged months ago, but only for the latest documentation. Does it need to be picked for 3.4, 3.3?

Ok, sorry. I only found it because I was having issues. Yes please pick for 3.3 and 3.4. I se now that it's in https://docs.godotengine.org/en/latest/tutorials/platform/ios/plugins_for_ios.html#game-center

**Comment by Calinou** [comment]
I can spot several issues in the current page on the `master` branch. For instance, The `in_app_store` variable is not used consistently.

**Pull request merged**